### PR TITLE
docs: update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by SEMIN-97**
## 변경사항
기존 CONTRIBUTING.md 가이드의 내용을 보강했습니다.
- #61 

## 체크리스트
- [ ] 기능 명세를 작성하였나요?
- [ ] 테스트 코드를 작성하였나요?

## 추가 논의사항
- 보강할 내용이 있을까요?
- 버그 제보용 이슈 템플릿을 제작하면 좋을 것같아요.
- 해당 문서와 커밋 메시지도 영어로 작성하고 있으니, 이슈/PR 템플릿과 컴포넌트 생성 CLI 문구도 영어로 통일해야할지 고민이 됩니다..

## Summary by CodeRabbit

- **문서**
	- 기여 가이드라인 문서를 전면 개선
	- 기여 프로세스에 대한 명확한 지침 추가
	- 개발, 컴포넌트 생성, 테스팅, Storybook 등 새로운 섹션 포함
	- 풀 리퀘스트 생성 및 브랜치 명명 규칙에 대한 상세 지침 추가
	- 이슈 및 기능 제안 보고 가이드라인 상세화
	- 문서 구조와 가독성 개선

### CODE REVIEW

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -1,18 +1,102 @@
 # Contributing to Sipe Design System
 
-We love your input! We want to make contributing to this project as easy and transparent as possible. This document outlines the process for contributing to our component library.
+Welcome to our project! We want to make contributing to this project as easy and transparent as possible.
+This document provides a clear guide on **how to contribute to Sipe Design System**.
```
Comment (hy57in) [inline .github/CONTRIBUTING.md:4]: 사소하지만 Side 이름을 쓰는 건 어때?
```suggestion
This document provides a clear guide on **how to contribute to Side(Sipe Design System)**.
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -1,18 +1,102 @@
...
+
+---
+
+## How to Propose Changes
+Before proposing changes, please **open an issue** to discuss the bug or feature.
+
+### Create a Pull Request
+
+1. **Create a Branch**  
+   Follow the branch naming convention: `<type>/<subject>`
```
[the remark this answers is not in the document]
Comment (hy57in) [reply .github/CONTRIBUTING.md:65]: 컨벤션 있으면 좋을것 같다고 생각하는데!

암묵적으로 `feat/<컴포넌트명>` 이렇게 쓰고 있는것 같아

feat, feature 등 여러 방식으로 만들 수 있으니 컨벤션이 정해진다면 추가하는게 좋겠어

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -1,18 +1,102 @@
...
+   ```sh
+   git checkout -b feat/your-feature
```
Comment (hy57in) [inline .github/CONTRIBUTING.md:67]: 항상 feat이 아닐 수 있으니 
```suggestion
   git checkout -b your-branch
```
이 정도는 어때?

**Review by hy57in — APPROVED** [review]
짱이다~~~

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (froggy1014) [reply .github/CONTRIBUTING.md:67]: branch도 컨벤션... 정해야하나 ?

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -1,18 +1,102 @@
...
+
+---
+
+## Issue Reporting
+
+We use **GitHub Issues** to track and manage problems and discussions.  
+Please register your issue so we can discuss and decide on the next steps.
 
-## Creating a New Component
+### Bug Reporting
```
[the remark this answers is not in the document]
Comment (SEMIN-97) [reply .github/CONTRIBUTING.md:97]: 버그 리포트 템플릿은 바로 만들어볼게!

File: .github/CONTRIBUTING.md [hunk 5]
```diff
@@ -1,18 +1,102 @@
...
+
+- **Naming Convention:** Use kebab-case for component names (e.g., `my-button`).
+- **Styling:** Prioritize using predefined styles from the `tokens` package.
+
+### Testing
+We use **Vitest** for testing.  
+All new features or changes must include **relevant test cases**.
+```sh
+pnpm test
+```
```
Comment (developerjhp) [inline .github/CONTRIBUTING.md:47]: 테스트의 최소조건 명시적으로 추가하는 건 어떨까?

**Minimum Testing Requirements:**
- Cover all primary use cases.
- Ensure edge cases are tested.

File: .github/CONTRIBUTING.md (hunk 1 above)
Comment (developerjhp) [reply .github/CONTRIBUTING.md:4]: Side가 무엇인지 용어 정의가 추가되면 더 좋을 것 같아!

**What is Side?**
Side refers to the Sipe Design System, our primary UI component library aimed at creating consistent and reusable interfaces.

File: .github/CONTRIBUTING.md (hunk 3 above)
Comment (SEMIN-97) [reply .github/CONTRIBUTING.md:64]: #86 이슈가 결정되면 컨벤션 내용 추가할게!

**Review by noahluftyang — APPROVED** [review]
